core/trino-main/src/main/java/io/trino/operator/aggregation/DecimalAverageAggregation.java

```diff
@@ -114,119 +114,140 @@ public AggregationMetadata specialize(BoundSignature boundSignature)
                 ImmutableList.of(new AccumulatorStateDescriptor<>(
                         stateInterface,
                         stateSerializer,
-                        new LongDecimalWithOverflowAndLongStateFactory())));
+                        new Int128StateFactory()),
+                        new AccumulatorStateDescriptor<>(
+                                stateInterface,
+                                stateSerializer,
+                                new Int128StateFactory())));
     }
 
-    public static void inputShortDecimal(LongDecimalWithOverflowAndLongState state, Block block, int position)
+    public static void inputShortDecimal(Int128State decimalState, Int128State counterOverflowState, Block block, int position)
```

Comment:
make it `Int128State decimalState`, `LongState counter` and `NullableLongState overflow`.
Usually `NullableLongState` will be all `0` so it makes sense to put `null` as `0`. In case all values are `null` such block will be serialized to RLE